src/Controller/Component/CheckHttpCacheComponent.php

```diff
@@ -36,7 +36,8 @@ class CheckHttpCacheComponent extends Component
     /**
      * Before Render hook
      *
-     * @param \Cake\Event\EventInterface $event The Controller.beforeRender event.
+     * @template TSubject of object
+     * @param \Cake\Event\EventInterface<TSubject> $event The Controller.beforeRender event.
```

Comment:
```suggestion
     * @param \Cake\Event\EventInterface<\Cake\Controller\Controller> $event The Controller.beforeRender event.
```

We know the type here, in the controller layer the subject for all the events is a `Controller` instance. Similarly in tables and behaviors it would be `Cake\ORM\Table`.